File: frontend/src/types/incident-report.ts
export interface IncidentReport {
  id: number;
  title: string;
  status: number; // 0=chờ xử lý, 1=đã xử lý, 2=từ chối
  roomId: number | null;
  roomName: string | null;
  reporterId: number;
  reporterName: string;
  reporterRole: string; // "STUDENT" | "TEACHER"
  handlerId: number | null;
  handlerName: string | null;
  createdAt: string;
}

File: backend/src/main/java/com/bachld/backend/model/IncidentReport.java
package com.bachld.backend.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Table;
import lombok.*;
import lombok.experimental.FieldDefaults;

@AllArgsConstructor
@Getter
@Setter
@Entity
@FieldDefaults(level = AccessLevel.PRIVATE)
@NoArgsConstructor
@Table(name = "incident_reports")
public class IncidentReport extends BaseEntity {

  String title;

  @Column(name = "room_id")
  Integer roomId;

  @Column(name = "reporter_id")
  Integer reporterId;

  @Column(name = "reporter_role", length = 20)
  String reporterRole;

  @Column(name = "handler_id")
  Integer handlerId;
}
